ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -35,32 +35,32 @@ class DataCollectionViewModel @Inject internal constructor(
     private val submissionRepository: SubmissionRepository,
     private val locationOfInterestHelper: LocationOfInterestHelper
 ) : AbstractViewModel() {
-    val submission: @Hot(replays = true) LiveData<Loadable<Submission>>
+    val submission: @Hot(replays = true) LiveData<Submission?>
     val jobName: @Hot(replays = true) LiveData<String>
     val loiName: @Hot(replays = true) LiveData<String>
 
     private val argsProcessor: @Hot(replays = true) FlowableProcessor<DataCollectionFragmentArgs> =
         BehaviorProcessor.create()
 
     init {
-        val submissionStream: Flowable<Loadable<Submission>> =
+        val submissionStream: Flowable<Submission?> =
             argsProcessor.switchMapSingle { args ->
                 submissionRepository.createSubmission(
                     args.surveyId, args.locationOfInterestId, args.submissionId
-                ).map { Loadable.loaded(it) }.onErrorReturn { Loadable.error(it) }
+                ).map { it }.onErrorReturn { null }
```

Comment:
Is `map()` operator required if we are not transforming the object? Same question for `onErrorReturn()`

What would happen if we remove both?

Reply:
Looks like we can remove them both. Thanks for the suggestion!

Reply:
Actually it looks like I get an error after removing the onErrorReturn so I added that back in.